physics/SFC_Models/Land/Noahmp/noahmpdrv.F90

```diff
@@ -9,99 +9,446 @@
+    public :: noahmpdrv_init, noahmpdrv_run, &
+              noahmpdrv_timestep_init, noahmpdrv_finalize
 
-      contains
+    !> \Land IAU data and control
+    ! Land IAU Control holds settings' information, maily read from namelist (e.g., block of global domain that belongs to a process ,
+    ! whether to do IAU increment at this time step, time step informatoin, etc)    
+    type (land_iau_control_type)          :: Land_IAU_Control
+    ! Land IAU Data holds spatially and temporally interpolated soil temperature increments per time step
+    type (land_iau_external_data_type)    :: Land_IAU_Data   !(number of blocks):each proc holds nblks
+
+    contains
 
 !> \ingroup NoahMP_LSM
 !! \brief This subroutine is called during the CCPP initialization phase and calls set_soilveg() to 
 !! initialize soil and vegetation parameters for the chosen soil and vegetation data sources.
 !! \section arg_table_noahmpdrv_init Argument Table
 !! \htmlinclude noahmpdrv_init.html
 !!
-      subroutine noahmpdrv_init(lsm, lsm_noahmp, me, isot, ivegsrc, &
-                                nlunit, pores, resid,               &
-                                do_mynnsfclay,do_mynnedmf,          &
-                                errmsg, errflg)
-
-        use machine,          only: kind_phys
-        use set_soilveg_mod,  only: set_soilveg
-        use namelist_soilveg
-        use noahmp_tables
+  subroutine noahmpdrv_init(lsm, lsm_noahmp, me,               & 
+                                isot, ivegsrc,                     &
+                                nlunit, pores, resid,              &
+                                do_mynnsfclay,do_mynnedmf,         &
+                                errmsg, errflg,                    &
+                                mpi_root,                                &
+                                fn_nml, input_nml_file, isc, jsc, ncols, nx, ny, tile_num, &
+                                nblks, blksz, xlon, xlat,                             &     
+                                lsoil, lsnow_lsm, dtp, fhour)
+
+    use machine,          only: kind_phys
+    use set_soilveg_mod,  only: set_soilveg
+    use namelist_soilveg
+    use noahmp_tables
+
+    implicit none
+
+    integer,              intent(in) :: lsm
+    integer,              intent(in) :: lsm_noahmp    
+    integer,              intent(in) :: me         !  mpi_rank
+    integer,              intent(in) :: isot, ivegsrc, nlunit
+    real (kind=kind_phys), dimension(:), intent(out) :: pores, resid
+    logical,              intent(in) :: do_mynnsfclay
+    logical,              intent(in) :: do_mynnedmf
+    character(len=*),     intent(out) :: errmsg
+    integer,              intent(out) :: errflg
+    ! land iau mod
+    integer,                       intent(in) :: mpi_root   ! = GFS_Control%master        
+    character(*),                  intent(in) :: fn_nml
+    character(len=:), intent(in), dimension(:) :: input_nml_file 
```

Comment:
This needs to be declared with the pointer attribute in order to compile. See https://github.com/tsga/ccpp-physics/pull/1